modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/DateFilter/index.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+import ClayButton from '@clayui/button';
+import ClayDatePicker from '@clayui/date-picker';
+import {useEffect, useState} from 'react';
+
+const DateFilter = ({
+	children,
+	clearInputs,
+	onOrAfterDisabled,
+	onOrBeforeDisabled,
+	updateFilters,
+}) => {
+	const [expandedOnOrAfter, setExpandedOnOrAfter] = useState(false);
+	const [expandedOnOrBefore, setExpandedOnOrBefore] = useState(false);
+
+	const [onOrAfterValue, setOnOrAfterValue] = useState('');
+	const [onOrBeforeValue, setOnOrBeforeValue] = useState('');
+
+	const now = new Date();
+
+	useEffect(() => {
+		if (onOrAfterDisabled) {
+			setOnOrAfterValue('');
+		}
+	}, [onOrAfterDisabled]);
+
+	useEffect(() => {
+		if (onOrBeforeDisabled) {
+			setOnOrBeforeValue('');
+		}
+	}, [onOrBeforeDisabled]);
+
+	useEffect(() => {
+		if (clearInputs) {
+			setOnOrAfterValue('');
+			setOnOrBeforeValue('');
+		}
+	}, [clearInputs]);
+
+	return (
+		<div className="p-3 w-100">
+			<div className="font-weight-semi-bold pb-3 text-paragraph">
+				On or after
+				<ClayDatePicker
+					dateFormat="MM/dd/yyyy"
+					disabled={onOrAfterDisabled}
+					expanded={expandedOnOrAfter}
+					onExpandedChange={setExpandedOnOrAfter}
+					onValueChange={(value, eventType) => {
+						setOnOrAfterValue(value);
+
+						if (eventType === 'click') {
+							setExpandedOnOrAfter(false);
+						}
+					}}
+					placeholder="MM/DD/YYYY"
+					value={onOrAfterValue}
+					years={{
+						end: now.getFullYear() + 5,
+						start: now.getFullYear() - 5,
```

Comment:
Assign the number to a variable. Who is 5?

Reply:
done!